src/Akka.Persistence.Redis/Query/Stages/AllEventsSource.cs

```diff
@@ -0,0 +1,348 @@
+                            {
+                                tuple.Item2 = pr.SequenceNr + 1;
+                            }
+
+                            return tuple;
+                        });
+
+                        _currentSequenceNr = maxSequenceNr;
+                        Log.Debug($"Max sequence number is now {maxSequenceNr}");
+                        if (evts.Count > 0)
+                        {
+                            evts.ForEach(_buffer.Enqueue);
+                            Deliver();
+                        }
+                        else
+                        {
+                            // requery immediately
+                            _state = State.Idle;
+                            Query();
+                        }
+                    }
+                });
+
+                if (_live)
+                {
+                    // subscribe to notification stream only if live stream was required
+                    var messageCallback = GetAsyncCallback<(RedisChannel channel, string bs)>(data =>
+                    {
+                        if (data.channel.ToString().StartsWith(_journalHelper.GetJournalChannel()))
+                        {
+                            Log.Debug("Message received");
+
+                            switch (_state)
+                            {
+                                case State.Idle:
+                                    // do nothing, no query is running and no client request was performed
+                                    break;
+                                case State.Querying:
+                                    _state = State.NotifiedWhenQuerying;
+                                    break;
+                                case State.NotifiedWhenQuerying:
+                                    // do nothing we already know that some new events may exist
+                                    break;
+                                case State.WaitingForNotification:
+                                    _state = State.Idle;
+                                    Query();
+                                    break;
+                            }
+                        }
+                        else
+                        {
+                            Log.Debug($"Message from unexpected channel: {data.channel}");
+                        }
+                    });
+
+                    _subscription = _redis.GetSubscriber();
+                    _subscription.Subscribe(_journalHelper.GetJournalChannel("*"), (channel, value) =>
+                    {
+                        messageCallback.Invoke((channel, value));
+                    });
+                }
+                else
+                {
+                    // start by first querying the current highest sequenceNr
+                    // for the given persistent id
+                    // stream will stop once this has been delivered
+                    _state = State.Initializing;
```

Comment:
Why is the state machine starts up as initializing if it is not live, but starts as idle when it is live when both path uses the same initialization code?